plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithuRepositoryData.tsx

```diff
@@ -26,14 +26,10 @@ export const useGithuRepositoryData = (url: string) => {
   return useAsync(async (): Promise<GithubRepositoryData> => {
     const token = await githubAuthApi.getAccessToken(['repo']);
 
-    const response = await new Octokit({
-      auth: token,
-      ...(baseUrl && { baseUrl }),
-    }).request({ url: url });
-
-    return {
-      htmlUrl: response.data.html_url,
-      fullName: response.data.full_name,
-    };
+    return new GithubPullRequestsClient().getRepositoryData({
```

Comment:
I am not sure about this :/ So I was needed this logic from this hook inside an async hook, so I took the logic and put it into the api so it could be used with the useApi() hook. However this hook is used in the homepage componenets, and in the homepage we do not have access to the api provider so actually we are not able to use it via that hook that's why I opted for this solution. 
I am unsure what drawbacks it can have with instantiating a new client on every hook call (I guess)

My other thought is maybe it could work if we setup the homepage a way that it could use the useApi() hook ? 
What do you think ?

Reply:
I think it's ok to construct the client locally for simplicity sake. Could be better to pull the initialization outside of the hook though. You could even possibly wrap that into a useMemo with an empty dep array or something but it's likely more overhead perf wise to do that than initializing it every time. The class itself is tiny so I wouldn't worry too much about it